text/1129-deprecated-computed-properties.md

```diff
@@ -0,0 +1,333 @@
+
+Also implicitly covered: use of `computed()` with dependency key arguments (classic pattern) and setter forms, plus `dependentKeyCompat` decorator.
+
+Out of scope (NOT deprecated by this RFC):
+- `@tracked` (unchanged)
+- Native getters / setter semantics
+- Class field / decorator proposals unrelated to CPs
+
+### General Migration Principles
+
+1. Replace `computed` macro-derived properties with native getters using tracked dependencies.
+2. Use standard JS methods (`Array.prototype.filter`, `map`, `some`, `every`, `includes`, spread syntax, `Set`, etc.) or minimal utility functions you write locally.
+3. Prefer explicit, readable logic rather than dense macro chains.
+4. For aliasing / indirection, refactor to simple getter forwarding or direct property access injection.
+5. For one-way / read-only semantics, rely on conventions, TypeScript readonly typings, or simple getters without setters.
+6. For mutable derivations once provided by macros that returned array proxies, compute on access within a getter or precompute and cache in a tracked property if performance critical.
+
+### General `computed()` Migration
+
+Classic signature examples slated for deprecation:
+
+```js
+fullName: computed('firstName', 'lastName', function () {
+  return `${this.firstName} ${this.lastName}`;
+}),
+
+hasErrors: computed('errors.[]', function () { /* ... */ }),
+
+sorted: computed('items.[]', 'sortDefinition.[]', function() { /* ... */ }),
+```
+
+Migration uses a native getter; dependencies become `@tracked` on the fields accessed.
+
+```js
+import { tracked } from '@glimmer/tracking';
+
+class UserModel {
+  @tracked firstName;
+  @tracked lastName;
+
+  get fullName() {
+    return `${this.firstName} ${this.lastName}`;
+  }
+}
+```
+
+Collection dependency keys like `items.[]`, `items.@each.name`, etc. are replaced by direct property access within the getter. Autotracking observes every `@tracked` read. If you push into an array (mutating method) you must reassign to trigger, e.g. `this.items = [...this.items, newItem]`. Codemod will wrap mutating calls or emit TODO comments.
+
+For expensive computations previously cached by computed's default memoization, you may cache manually:
+
+```js
+get expensiveResult() {
+  if (this._cachedExpensive === undefined || this._cacheKey !== this.inputVersion) {
+    this._cachedExpensive = doWork(this.inputs);
+    this._cacheKey = this.inputVersion;
+  }
+  return this._cachedExpensive;
+}
+```
+
+In practice, most computed properties are inexpensive and can just be recomputed each access.
+
+### Macro-by-Macro Replacement Patterns
+
+Below each macro lists: Purpose, Before, After (tracked + JS), and Codemod Strategy.
```

Comment:
FYI There is already a codemod available. I built one a while ago to completely migrate a codebase off macros: https://github.com/OTA-Insight/ember-computed-macro-codemod/

It does not cover all macro's though, so there's room to extend it.